src/backend/aspen/database/models/sample.py

```diff
@@ -253,6 +253,22 @@ class Sample(idbase, DictMixin):  # type: ignore
     sequencing_reads_collection: Optional[SequencingReadsCollection]
     uploaded_pathogen_genome: Optional[UploadedPathogenGenome]
 
+    def generate_public_identifier(self):
```

Comment:
This is one of the more "interesting" bits of code in this PR. If someone doesn't send us a public identifier, we generate that identifier via invoking SQL functions, like so:
```INSERT INTO SAMPLES (..., public_identifier, ...) VALUES (...,  CONCAT("hCov-19/USA/CADPH-", currval('aspen.samples_id_seq'), "/2022")```

We do this so we don't have to pre-generate DB identifiers outside a transaction, with nothing guarantees that they'll match the row ID's, like the previous code did.